### PR TITLE
Fix #9726: Update mixer on replace instrument

### LINKED ISSUE #9726: [MU4 Issue] Part name on Mixer doesn't change
**Describe the bug**
When editing the part name from the Instruments panel, it doesn't change the name on the mixer either. 

https://user-images.githubusercontent.com/71656189/141502466-3a4bcbf7-7693-4274-9229-9cf013434bf8.mp4

**Desktop (please complete the following information):**
 - OS: Windows and MacOS

### PR DESCRIPTION
**Pull request by HemantAntony**
Resolves: #9726 

This commit updates the mixer when an instrument is replaced.

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [ ] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by HemantAntony** [comment]
In my opinion, this is a temporary fix. A better fix should be made. Though this appears to work, the order does change. So the new instrument is last

File: src/playback/internal/playbackcontroller.cpp [hunk 1]
```diff
@@ -600,6 +600,11 @@ void PlaybackController::setupSequenceTracks()
     partList.onItemRemoved(this, [this](const Part* part) {
         removeTrack(part->id());
     });
+
+    partList.onItemChanged(this, [this](const Part* part) {
+        addTrack(part->id(), part->partName().toStdString());
+        removeTrack(part->id());
```
Comment (Jojo-Schmitz) [inline src/playback/internal/playbackcontroller.cpp:606]: Isn't that the wrong way round and should remove first and add last?

Comment (HemantAntony) [reply src/playback/internal/playbackcontroller.cpp:606]: Yes, that's what doesn't make sense. The other way doesn't work. When you do it the other way, the instrument gets removed, but the last instrument in the mixer gets duplicated

**Comment by RomanPudashkin** [comment]
I don't think this fix is reliable enough. Let's close this PR to give other developers a chance to work on this bug

**Pull request closed without merging**